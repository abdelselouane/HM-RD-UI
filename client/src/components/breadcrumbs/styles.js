import styled from 'styled-components';

export const Container = styled.div`
  grid-area: cmess;

  .breadcrumb {
    font-size: 16px;
  }
`;

export default Container;
