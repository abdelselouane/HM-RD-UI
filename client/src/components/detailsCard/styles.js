import styled from 'styled-components';

export const Card = styled.div`
  width: 85%;
  margin-bottom: 20px;

  .card-content {
    padding: 0px 25px;
  }
`;

export const Title = styled.span`
  align-items: flex-end;
  display: flex;
  justify-content: flex-start;
  width: 100%;
  border-bottom: solid #666 1px;
  padding-bottom: 15px;

  h2 {
    display: inline-block;
    color: #f96302;
    font-weight: 900;
    padding-right: 25px;
  }

  span {
    text-transform: capitalize;
  }
`;

export const Row = styled.div`
  display: flex;
  padding: 15px;
`;

export const Col = styled.div`
  ${props => props.flex && `flex: ${props.flex}`}
  display: flex;
  flex-direction: column;
`;

export const Label = styled.div`
  font-weight: 700;
  color: #666;
`;
