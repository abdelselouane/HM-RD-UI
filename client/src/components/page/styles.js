import styled, { css } from 'styled-components';

const defaultGrid = css`
  display: grid;
  grid-template-rows: 70px
                      48px
                      24px
                      24px
                      repeat(3, 1fr)
                      repeat(1, 72px);
  grid-template-columns: 20px 1fr repeat(4, 3fr) 1fr 20px;
  grid-template-areas:  ". . . . . . . ."
                        ". . . . . . . ."
                        ". . cmess cmess cmess cmess . ."
                        ". . cmess cmess cmess cmess . ."
                        ". . carea carea carea carea . ."
                        ". . carea carea carea carea . ."
                        ". . carea carea carea carea . ."
                        ". . . . . . . .";
`;

const lowResolutionGrid = css`
  display: grid;
  grid-template-areas:  ". . . . . . . ."
                        ". . . . . . . ."
                        ". cmess cmess cmess cmess cmess cmess ."
                        ". cmess cmess cmess cmess cmess cmess ."
                        ". carea carea carea carea carea carea ."
                        ". carea carea carea carea carea carea ."
                        ". carea carea carea carea carea carea ."
                        ". . . . . . . .";
  grid-template-rows: 70px
                      24px
                      24px
                      24px
                      repeat(3, 1fr)
                      repeat(1, 48px);
  grid-template-columns: 10px 1fr repeat(4, 3fr) 1fr 10px;
`;

export const GridLayout = styled.div`
  background-color: ${({ theme }) => theme.black12};
  box-sizing: border-box;
  color: ${({ theme }) => theme.blackA11};
  height: 100%;  
  ${props => (props.enableScroll ? 'overflow-x: scroll;' : 'overflow: hidden;')}
  
  ${() => defaultGrid}
  @media only screen and (max-width: 1024px) {
    ${() => lowResolutionGrid}
  }
`;

export const Container = styled.div`
  align-items: flex-start;
  display: flex;
  grid-area: carea;
  justify-content: center;
  ${({ hideBoxShadow }) => (hideBoxShadow ? '' : 'box-shadow: 0 0 2px 0 rgba(0, 0, 0, 0.12), 0 2px 2px 0 rgba(0, 0, 0, 0.24);')}
`;
