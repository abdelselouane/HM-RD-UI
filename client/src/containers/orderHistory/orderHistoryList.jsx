import React, { Component } from 'react';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

import { Container } from './styles';

import * as PartsOrdersActions from '../../actions/partsOrdersActions';

import Page from '../../components/page';
import Table from '../../components/table';

import { withContext } from '../../contexts/history';

import { getFormattedDate } from '../../utils/formatHelper';

export class OrderHistoryList extends Component {
  static propTypes = {
    actions: PropTypes.shape({
      getPartsOrders: PropTypes.func.isRequired
    }).isRequired,
    history: PropTypes.shape({
      push: PropTypes.func.isRequired
    }).isRequired,
    orders: PropTypes.arrayOf(
      PropTypes.shape({
        orderCreatedDate: PropTypes.string.isRequired,
        partsOrderId: PropTypes.number.isRequired,
        workOrderNbr: PropTypes.string.isRequired,
        partsDetails: PropTypes.arrayOf(PropTypes.shape({})).isRequired
      })
    ).isRequired
  };

  componentWillMount() {
    this.props.actions.getPartsOrders();
  }

  getRows = orders =>
    orders.map(({ partsDetails, ...order }) => ({
      key: order.partsOrderId,
      orderCreatedDate: getFormattedDate(order.orderCreatedDate),
      partsOrderId: order.partsOrderId,
      workOrderNbr: order.workOrderNbr || <em>Not Provided</em>,
      orderDesc: partsDetails
        .reduce((acc, { partDescription }) => `${acc}, ${partDescription}`, '')
        .substring(2)
    }));

  handleRowClick = ({ partsOrderId }) => {
    if (partsOrderId) {
      this.props.history.push(`orderHistory/${partsOrderId}`);
    }
  };

  columns = [
    {
      key: 'orderCreatedDate',
      name: 'Date',
      thProps: {
        paddingLeft: '40px',
        width: '15%'
      },
      tdProps: {
        paddingLeft: '40px',
        width: '15%'
      }
    },
    {
      key: 'partsOrderId',
      name: 'Order Number',
      thProps: {
        width: '15%'
      },
      tdProps: {
        width: '15%'
      }
    },
    {
      key: 'orderDesc',
      name: 'Order Description',
      useTitle: true,
      tdProps: {
        overflow: 'hidden',
        whiteSpace: 'nowrap',
        textOverflow: 'ellipsis'
      }
    },
    {
      key: 'workOrderNbr',
      name: 'Work Order Number',
      thProps: {
        width: '20%'
      },
      tdProps: {
        width: '20%'
      }
    }
  ];

  render() {
    return (
      <Page className="order-history" showHomeNav>
        <Container>
          <div className="title">
            <h2>Order History</h2>
          </div>
          <Table
            columns={this.columns}
            rows={this.getRows(this.props.orders)}
            showResults={this.props.orders.length > 0}
            handleRowClick={this.handleRowClick}
          />
        </Container>
      </Page>
    );
  }
}

const mapStateToProps = state => ({
  orders: state.partsOrders.orders
});

const mapDispatchToProps = dispatch => ({
  actions: bindActionCreators(PartsOrdersActions, dispatch)
});

const OrderHistoryListWithContext = withContext(OrderHistoryList);

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(OrderHistoryListWithContext);
