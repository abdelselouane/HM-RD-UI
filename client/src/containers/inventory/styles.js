import { Container as BaseContainer } from '../../components/page';

export const Container = BaseContainer.extend`
background-color: ${({ theme }) => theme.white};
box-sizing: border-box;
display: flex;
flex-direction: column;
justify-content: flex-start;
height: 100%;
overflow-y: hidden;
width: 100%;


.inventory-tabs {
  height: 100%;
}

.tabs span {
  &:focus {
    outline: none !important;
  }
  
  &[role=tab]:first-child {
    border-left: none;
  }
}

[role=tabpanel] {
    display: flex;
    flex-direction: column;
    height: 95%;
    overflow: hidden;
    padding: 20px 0px;

    .text-input-container.search {
      margin: 15px 15px;
      width: 65%;

      &.dropdown {
        width: 75%;
        margin: 15px 0px;
      }
    }
}

.search-container{
  display: flex;
  flex-direction: row;
  justify-content: space-between;

  button {
    flex: 0.15;
  }
}

.text-input-container .input-field {
  margin-top: 0em;
}

.edit-col .button {
  margin: 0px;
}

.text-input-container.search {
  flex: 0.5;

  input {
    margin-top: -6px;
  }

  .icon_search {
    bottom: 0;
    color: ${({ theme }) => theme.orange};
    font-size: 30px;
    padding: 5px 10px;
    position: absolute;
    right: 0;
    z-index: 1;
  }
}
`;

export default Container;
