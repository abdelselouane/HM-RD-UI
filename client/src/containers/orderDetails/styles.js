import styled from 'styled-components';
import { Container as BaseContainer } from '../../components/page';

export const Container = BaseContainer.extend`
  background-color: ${({ theme }) => theme.white};
  align-items: flex-start;
  display: flex;
  flex-direction: column;
`;

export const CardTitle = styled.div`
  align-items: flex-end;
  display: flex;
  justify-content: space-between;
  height: 50px;
  margin: 0px;
  margin-bottom: 30px;
  padding: 0px 20px;
  width: 100%;

  h2 {
    display: inline-block;
    margin-bottom: 0px;
  }
`;

export const PurchaseOrderList = styled.div`
  align-items: center;
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  padding-top: 20px;
  flex: 1;
  overflow-y: scroll;
  overflow-x: hidden;
  width: 100%;
`;
