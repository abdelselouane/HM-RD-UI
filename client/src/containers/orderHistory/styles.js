import { Container as BaseContainer } from '../../components/page';

export const Container = BaseContainer.extend`
  background-color: ${({ theme }) => theme.white};
  align-items: flex-start;
  display: flex;
  flex-direction: column;

  .title {
    padding: 0px 20px;
  }
  
  .order-history-header {
    margin: 0px;
  
    thead tr {
      margin-left: 1em;
  
      th {
        border: none;
        font-size: 18px;
  
        &:first-child {
          padding-left: 20px;
        }
      }
    }
  }
`;

export default Container;
