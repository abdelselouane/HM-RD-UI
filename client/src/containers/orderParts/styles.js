import { Container as BaseContainer } from '../../components/page';

export const Container = BaseContainer.extend`
background: ${({ theme }) => theme.white};
`;

export default Container;
