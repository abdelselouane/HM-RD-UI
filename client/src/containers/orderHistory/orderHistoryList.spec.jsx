/* global describe, beforeAll, it, expect, jest */
import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import * as Styles from './styles';

import { OrderHistoryList } from './orderHistoryList';

describe('<OrderHistoryList />', () => {
  let wrapper;
  let props;

  beforeAll(() => {
    props = {
      actions: {
        getPartsOrders: jest.fn()
      },
      history: {
        push: jest.fn()
      },
      orders: [
        {
          orderCreatedDate: '1970-12-02',
          partsOrderId: 2,
          workOrderNbr: '',
          partsDetails: [
            {
              partDescription: 'mockDescription3'
            }
          ]
        },
        {
          orderCreatedDate: '1970-12-01',
          partsOrderId: 1,
          workOrderNbr: '123456789',
          partsDetails: [
            {
              partDescription: 'mockDescription1'
            },
            {
              partDescription: 'mockDescription2'
            }
          ]
        }
      ]
    };

    wrapper = shallow(<OrderHistoryList {...props} />);
  });

  it('should render properly', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should call getPartsOrder on componentWillMount', () => {
    const spy = jest.spyOn(props.actions, 'getPartsOrders');
    wrapper.instance().componentWillMount();
    expect(spy).toHaveBeenCalled();
    spy.mockRestore();
    spy.mockReset();
  });

  it('should return rows in getRows call', () => {
    const rows = wrapper.instance().getRows(props.orders);
    expect(rows).toMatchObject([
      {
        key: 2,
        orderCreatedDate: 'Dec 02, 1970',
        partsOrderId: 2,
        workOrderNbr: <em>Not Provided</em>,
        orderDesc: 'mockDescription3'
      },
      {
        key: 1,
        orderCreatedDate: 'Dec 01, 1970',
        partsOrderId: 1,
        workOrderNbr: '123456789',
        orderDesc: 'mockDescription1, mockDescription2'
      }
    ]);
  });

  it('should call history.push to order details on handleRowClick', () => {
    const spy = jest.spyOn(props.history, 'push');
    const partsOrderId = 'mock';

    wrapper.instance().handleRowClick({});
    expect(spy).not.toHaveBeenCalled();

    wrapper.instance().handleRowClick({ partsOrderId });
    expect(spy).toHaveBeenCalledWith('orderHistory/mock');
    spy.mockRestore();
    spy.mockReset();
  });

  it('should render styles properly', () => {
    Object.values(Styles).forEach(Component => {
      expect(toJson(shallow(<Component />))).toMatchSnapshot();
    });
  });
});
