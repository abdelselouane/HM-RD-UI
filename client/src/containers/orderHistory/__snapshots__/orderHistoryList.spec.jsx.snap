// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OrderHistoryList /> should render properly 1`] = `
<Connect(withContext(Page))
  className="order-history"
  showHomeNav={true}
>
  <styled.div>
    <div
      className="title"
    >
      <h2>
        Order History
      </h2>
    </div>
    <Table
      className=""
      clearOnHover={false}
      columns={
        Array [
          Object {
            "key": "orderCreatedDate",
            "name": "Date",
            "tdProps": Object {
              "paddingLeft": "40px",
              "width": "15%",
            },
            "thProps": Object {
              "paddingLeft": "40px",
              "width": "15%",
            },
          },
          Object {
            "key": "partsOrderId",
            "name": "Order Number",
            "tdProps": Object {
              "width": "15%",
            },
            "thProps": Object {
              "width": "15%",
            },
          },
          Object {
            "key": "orderDesc",
            "name": "Order Description",
            "tdProps": Object {
              "overflow": "hidden",
              "textOverflow": "ellipsis",
              "whiteSpace": "nowrap",
            },
            "useTitle": true,
          },
          Object {
            "key": "workOrderNbr",
            "name": "Work Order Number",
            "tdProps": Object {
              "width": "20%",
            },
            "thProps": Object {
              "width": "20%",
            },
          },
        ]
      }
      handleRowClick={[Function]}
      handleSortClick={[Function]}
      isAlternating={true}
      isExpandable={false}
      rows={
        Array [
          Object {
            "key": 2,
            "orderCreatedDate": "Dec 02, 1970",
            "orderDesc": "mockDescription3",
            "partsOrderId": 2,
            "workOrderNbr": <em>
              Not Provided
            </em>,
          },
          Object {
            "key": 1,
            "orderCreatedDate": "Dec 01, 1970",
            "orderDesc": "mockDescription1, mockDescription2",
            "partsOrderId": 1,
            "workOrderNbr": "123456789",
          },
        ]
      }
      showBody={true}
      showCheckbox={false}
      showHeader={true}
      showResults={true}
    />
  </styled.div>
</Connect(withContext(Page))>
`;

exports[`<OrderHistoryList /> should render styles properly 1`] = `
.c0 {
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  grid-area: carea;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  box-shadow: 0 0 2px 0 rgba(0,0,0,0.12),0 2px 2px 0 rgba(0,0,0,0.24);
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c0 .title {
  padding: 0px 20px;
}

.c0 .order-history-header {
  margin: 0px;
}

.c0 .order-history-header thead tr {
  margin-left: 1em;
}

.c0 .order-history-header thead tr th {
  border: none;
  font-size: 18px;
}

.c0 .order-history-header thead tr th:first-child {
  padding-left: 20px;
}

<div
  className="c0"
/>
`;

exports[`<OrderHistoryList /> should render styles properly 2`] = `
.c0 {
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  grid-area: carea;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  box-shadow: 0 0 2px 0 rgba(0,0,0,0.12),0 2px 2px 0 rgba(0,0,0,0.24);
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c0 .title {
  padding: 0px 20px;
}

.c0 .order-history-header {
  margin: 0px;
}

.c0 .order-history-header thead tr {
  margin-left: 1em;
}

.c0 .order-history-header thead tr th {
  border: none;
  font-size: 18px;
}

.c0 .order-history-header thead tr th:first-child {
  padding-left: 20px;
}

<div
  className="c0"
/>
`;
